packages/ag-charts-community/src/chart/interaction/interactionManager.ts

```diff
@@ -167,6 +190,37 @@ export class InteractionManager extends BaseManager<InteractionTypes, Interactio
         });
     }
 
+    private makeRegion(name: RegionName): InteractionRegion {
+        return {
+            name,
+            listeners: new Listeners<InteractionTypes, (event: InteractionEvent<InteractionTypes>) => void>(),
+        };
+    }
+
+    public addRegion(name: RegionName, node: BBoxProvider): ObservableRegion {
```

Comment:
Yeah I'd agree with this, it would be good to keep the interaction manager focused on just simple interaction events as much as possible.

Reply:
ACK.

Reply:
I can move some functionality into a new `RegionManager` class, but the internal wrapper class in `InteractionManager.addRegion` needs to stay because it needs to know the exact type of the handler. Moving that wrapper class into `RegionManager` would require us to move types like `InteractionTypes` and `InteractionHandler<T>` into a new file to avoid circular imports.